server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -577,6 +581,10 @@ public ConnectionManager getConnectionManager() {
         return connectionManager;
     }
 
+    public TransportVersion getMinTransportVersion() {
+        return transportVersions.values().stream().min(Comparator.naturalOrder()).orElse(TransportVersion.MINIMUM_COMPATIBLE);
```

Comment:
I'm not sure what the fallback here is if there are no other connections; what do we do for single nodes here? What about single nodes that then become a multi-versioned cluster?